### ISSUE TITLE
Invalid ktlint:standard:wrapping error for higher-order function

### ISSUE THREAD

**Issue by dondod** [pinterest/ktlint#2165]
This is an odd case where a higher-order function with a signature like `function -> function -> <type> = { some-lambda }` results in a ktlint error, where it thinks the correct formatting involves indenting a line in a strange way. 

## Expected Behavior
No lint error

## Observed Behavior
Lint error: `Missing newline after "{" (standard:wrapping)`

## Steps to Reproduce
Run ktlint against this code (which is how IntelliJ formats the code):
```kotlin
class Foo {
    private val ktlintIssue: (String) -> (Int) -> Unit = { name ->
        { foo ->
            println("name = $name, foo = $foo")
        }
    }
}
```

Ktlint formats this as:

```Kotlin
class Foo {
    private val ktlintIssue: (String) -> (Int) -> Unit = { name ->
        {
                foo ->
            println("name = $name, foo = $foo")
        }
    }
}
```

To suppress this, it requires disabling wrapping and curly-spacing.  Disabling wrapping alone results in the curly-spacing error.
```kotlin
@Suppress("ktlint:standard:wrapping", "ktlint:standard:curly-spacing")
```
## Your Environment

* Version of ktlint used: 0.50.0
* Relevant parts of the `.editorconfig` settings
```
[*.{kt,kts}]
ktlint_code_style = ktlint_official
```
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):
Using custom code based on https://pinterest.github.io/ktlint/0.47.1/install/integrations/#custom-gradle-integration

```groovy
    tasks.register("ktlintFix", JavaExec) {
        group = "verification"
        description = "Check and fix Kotlin code style."
        classpath = configurations.ktlint
        mainClass = "com.pinterest.ktlint.Main"
        jvmArgs "--add-opens=java.base/java.lang=ALL-UNNAMED"
        args "src/**/*.kt", "**.kts", "-F"
    }

```
* Version of Gradle used (if applicable): 7.6
* Operating System and version: OSX 13.3.1

**Comment by paul-dingemans** [comment]
Duplicate of #2103 which is already fixed in snapshot version. Will be released in `1.0`
